src/embedding/common.ts

```diff
@@ -0,0 +1,129 @@
+// (C) 2020 GoodData Corporation
+
+import isEmpty = require('lodash/isEmpty');
+
+export enum GdcProductName {
+    ANALYTICAL_DESIGNER = 'analyticalDesigner',
+    KPI_DASHBOARDS = 'dashboards'
+}
+
+export enum GdcEventType {
+    AppCommandFailed = 'appCommandFailed'
+}
+
+export interface IGdcMessageEventData<Product, T, TBody> {
+    readonly product: Product;
+    readonly event: {
+        readonly name: T;
+        readonly data?: TBody;
+        readonly contextId?: string;
+    };
+}
+
+export interface IGdcMessageEventDataWrapper<Product, T, TBody> {
+    readonly gdc: IGdcMessageEventData<Product, T, TBody>;
+}
+
+export interface IGdcMessageEvent<Product, T, TBody> extends MessageEvent {
+    readonly data: IGdcMessageEventDataWrapper<Product, T, TBody>;
+}
+
+export type GdcMessageEventListener = (
+    event: IGdcMessageEvent<string, string, any>
```

Comment:
should we replace `any` with `TBody `?

Reply:
`GdcMessageEventListener` used in this file on file only, so don't need to generic this type